### PR TITLE
Semgrep: Add support for INFO severity

### PR DESCRIPTION
**Pull request by arthur-wolf**
Updated semgrep parser.py because the Info severity tag was not recognised

### CODE REVIEW

File: dojo/tools/semgrep/parser.py [hunk 1]
```diff
@@ -67,5 +67,7 @@ def convert_severity(self, val):
             return "Low"
         elif "ERROR" == val.upper():
             return "High"
+        elif "INFO" == val.upper():
+            return "Informational"
```
Comment (damiencarol) [inline dojo/tools/semgrep/parser.py:71]: `Informational` is not a correct value. we should have `Info` here.
```suggestion
            return "Info"
```

**Review by damiencarol — CHANGES_REQUESTED** [review]

**Comment by damiencarol** [comment]
@VeepsFR do you have a report that fail to import? What is the need of this PR?

**Comment by VeepsFR** [comment]
@damiencarol I tried to import a JSON report and obviously could not since the Info tag was not taken into account in the code, thus the PR.

**Comment by Maffooch** [comment]
I believe Damien is asking if you could add a unit test with the redacted JSON report that failed the import and add it here https://github.com/DefectDojo/django-DefectDojo/tree/master/dojo/unittests/scans/semgrep so that issues like these are less frequent.

**Comment by shizzgar** [comment]
+1 here.

Dojo uwsgi logs:

```
uwsgi_1         | [20/Jul/2021 14:29:38] ERROR [dojo.importers.utils:23] Unknown value for severity: INFO
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/app/./dojo/importers/utils.py", line 16, in parse_findings
uwsgi_1         |     parsed_findings = parser.get_findings(scan, test)
uwsgi_1         |   File "/app/./dojo/tools/semgrep/parser.py", line 26, in get_findings
uwsgi_1         |     severity=self.convert_severity(item["extra"]["severity"]),
uwsgi_1         |   File "/app/./dojo/tools/semgrep/parser.py", line 71, in convert_severity
uwsgi_1         |     raise ValueError(f"Unknown value for severity: {val}")
uwsgi_1         | ValueError: Unknown value for severity: INFO
uwsgi_1         | [20/Jul/2021 14:29:38] WARNING [dojo.importers.utils:24] Error in parser: Unknown value for severity: INFO
uwsgi_1         | [20/Jul/2021 14:29:38] ERROR [django.request:224] Internal Server Error: /api/v2/import-scan/
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/app/./dojo/api_v2/serializers.py", line 1228, in save
uwsgi_1         |     test, finding_count, closed_finding_count = importer.import_scan(scan, scan_type, engagement, lead, environment,
uwsgi_1         |   File "/app/./dojo/importers/importer/importer.py", line 290, in import_scan
uwsgi_1         |     parsed_findings = importer_utils.parse_findings(scan, test, active, verified, scan_type)
uwsgi_1         |   File "/app/./dojo/importers/utils.py", line 16, in parse_findings
uwsgi_1         |     parsed_findings = parser.get_findings(scan, test)
uwsgi_1         |   File "/app/./dojo/tools/semgrep/parser.py", line 26, in get_findings
uwsgi_1         |     severity=self.convert_severity(item["extra"]["severity"]),
uwsgi_1         |   File "/app/./dojo/tools/semgrep/parser.py", line 71, in convert_severity
uwsgi_1         |     raise ValueError(f"Unknown value for severity: {val}")
uwsgi_1         | ValueError: Unknown value for severity: INFO
uwsgi_1         | 
uwsgi_1         | During handling of the above exception, another exception occurred:
uwsgi_1         | 
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
uwsgi_1         |     response = get_response(request)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
uwsgi_1         |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
uwsgi_1         |     return view_func(*args, **kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/viewsets.py", line 125, in view
uwsgi_1         |     return self.dispatch(request, *args, **kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 509, in dispatch
uwsgi_1         |     response = self.handle_exception(exc)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 469, in handle_exception
uwsgi_1         |     self.raise_uncaught_exception(exc)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
uwsgi_1         |     raise exc
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 506, in dispatch
uwsgi_1         |     response = handler(request, *args, **kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/mixins.py", line 19, in create
uwsgi_1         |     self.perform_create(serializer)
uwsgi_1         |   File "/app/./dojo/api_v2/views.py", line 1869, in perform_create
uwsgi_1         |     serializer.save(push_to_jira=push_to_jira)
uwsgi_1         |   File "/app/./dojo/api_v2/serializers.py", line 1243, in save
uwsgi_1         |     raise Exception(ve)
uwsgi_1         | Exception: Unknown value for severity: INFO
uwsgi_1         | ERROR:django.request:Internal Server Error: /api/v2/import-scan/
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/app/./dojo/api_v2/serializers.py", line 1228, in save
uwsgi_1         |     test, finding_count, closed_finding_count = importer.import_scan(scan, scan_type, engagement, lead, environment,
uwsgi_1         |   File "/app/./dojo/importers/importer/importer.py", line 290, in import_scan
uwsgi_1         |     parsed_findings = importer_utils.parse_findings(scan, test, active, verified, scan_type)
uwsgi_1         |   File "/app/./dojo/importers/utils.py", line 16, in parse_findings
uwsgi_1         |     parsed_findings = parser.get_findings(scan, test)
uwsgi_1         |   File "/app/./dojo/tools/semgrep/parser.py", line 26, in get_findings
uwsgi_1         |     severity=self.convert_severity(item["extra"]["severity"]),
uwsgi_1         |   File "/app/./dojo/tools/semgrep/parser.py", line 71, in convert_severity
uwsgi_1         |     raise ValueError(f"Unknown value for severity: {val}")
uwsgi_1         | ValueError: Unknown value for severity: INFO
uwsgi_1         | 
uwsgi_1         | During handling of the above exception, another exception occurred:
uwsgi_1         | 
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
uwsgi_1         |     response = get_response(request)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
uwsgi_1         |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
uwsgi_1         |     return view_func(*args, **kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/viewsets.py", line 125, in view
uwsgi_1         |     return self.dispatch(request, *args, **kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 509, in dispatch
uwsgi_1         |     response = self.handle_exception(exc)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 469, in handle_exception
uwsgi_1         |     self.raise_uncaught_exception(exc)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
uwsgi_1         |     raise exc
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 506, in dispatch
uwsgi_1         |     response = handler(request, *args, **kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.8/site-packages/rest_framework/mixins.py", line 19, in create
uwsgi_1         |     self.perform_create(serializer)
uwsgi_1         |   File "/app/./dojo/api_v2/views.py", line 1869, in perform_create
uwsgi_1         |     serializer.save(push_to_jira=push_to_jira)
uwsgi_1         |   File "/app/./dojo/api_v2/serializers.py", line 1243, in save
uwsgi_1         |     raise Exception(ve)
uwsgi_1         | Exception: Unknown value for severity: INFO
```
json report from semgrep:

```
{"results": [{"check_id": "rules.contrib.owasp.java.ssrf.owasp.java.ssrf.possible.import.statements", "end": {"col": 20, "line": 7}, "extra": {"is_ignored": false, "lines": "import java.net.URI;", "message": "Non-exhaustive list of Libraries that provide functionality to accept URL as a parameter\n", "metadata": {"category": "security", "cwe": "CWE-918: Server-Side Request Forgery (SSRF)", "source-rule-url": "https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html"}, "metavars": {}, "severity": "INFO"}, "path": "src/main/java/ru/itfbgroup/mvc/form/canonic/ReflectionUtils.java", "start": {"col": 1, "line": 7}}, {"check_id": "rules.contrib.owasp.java.ssrf.owasp.java.ssrf.possible.import.statements", "end": {"col": 20, "line": 8}, "extra": {"is_ignored": false, "lines": "import java.net.URL;", "message": "Non-exhaustive list of Libraries that provide functionality to accept URL as a parameter\n", "metadata": {"category": "security", "cwe": "CWE-918: Server-Side Request Forgery (SSRF)", "source-rule-url": "https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html"}, "metavars": {}, "severity": "INFO"}, "path": "src/main/java/ru/itfbgroup/mvc/form/canonic/ReflectionUtils.java", "start": {"col": 1, "line": 8}}], "errors": []}
```

When could i expect this merge in release?)

**Review by arthur-wolf — COMMENTED** [review]
@damiencarol In my opinion this is now correct and can be approved for merge.

**Review by Maffooch — CHANGES_REQUESTED** [review]
@arthur-wolf please add a new unit test to https://github.com/DefectDojo/django-DefectDojo/tree/master/dojo/unittests/scans/semgrep that tests the new change of this PR

**Comment by valentijnscholten** [comment]
or allow us write permission to the PR so we can update it, my push was denied.

**Comment by valentijnscholten** [comment]
hmm might be `gh cli` messing up branch names

**Comment by valentijnscholten** [comment]
diff to apply:
```
diff --git a/dojo/unittests/scans/semgrep/many_findings.json b/dojo/unittests/scans/semgrep/many_findings.json
index e37c7449c..08ccae9b4 100644
--- a/dojo/unittests/scans/semgrep/many_findings.json
+++ b/dojo/unittests/scans/semgrep/many_findings.json
@@ -130,7 +130,7 @@
             "https://capec.mitre.org/data/definitions/463.html"
           ]
         },
-        "severity": "WARNING",
+        "severity": "INFO",
         "fix": "javax crypto Cipher.getInstance(\"AES/GCM/NoPadding\");",
         "lines": "\t\t\tjavax.crypto.Cipher c = javax.crypto.Cipher.getInstance(\"DES/CBC/PKCS5Padding\");"
       }
diff --git a/dojo/unittests/tools/test_semgrep_parser.py b/dojo/unittests/tools/test_semgrep_parser.py
index e6b421b62..916d47839 100644
--- a/dojo/unittests/tools/test_semgrep_parser.py
+++ b/dojo/unittests/tools/test_semgrep_parser.py
@@ -40,7 +40,7 @@ class TestSemgrepParser(TestCase):
         self.assertEqual("javax crypto Cipher.getInstance(\"AES/GCM/NoPadding\");", finding.mitigation)
         self.assertEqual("java.lang.security.audit.cbc-padding-oracle.cbc-padding-oracle", finding.vuln_id_from_tool)
         finding = findings[2]
-        self.assertEqual("Low", finding.severity)
+        self.assertEqual("Info", finding.severity)
         self.assertEqual("src/main/java/org/owasp/benchmark/testcode/BenchmarkTest01150.java", finding.file_path)
         self.assertEqual(66, finding.line)
         self.assertEqual(696, finding.cwe)
```

**Comment by valentijnscholten** [comment]
Closed by accident @arthur-wolf ?

**Comment by arthur-wolf** [comment]
@valentijnscholten Tell me how I can allow you write permission. I don't know how to setup a unit test since I discovered this during an internship before starting my studies in IT. My coworkers have been telling me that it might be a lot to do compared to how little my change was (I mean, 2 lines of code ???).
I have a lot of stuff to do so if I can let you continue with that, I will with great pleasure !

**Comment by valentijnscholten** [comment]
https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

It's just two small changes as you can see in the diff I posted. We have so many errors on mistakes in parsers, do we need all the unit tests we can get.

**Comment by damiencarol** [comment]
@arthur-wolf if you can provide a report that have this severity data and let us commit on your branch we can help finish it.

**Comment by arthur-wolf** [comment]
Here is a scan report with a "INFO" Severity at line 16135 
[scan-js-2-clean.zip](https://github.com/DefectDojo/django-DefectDojo/files/7074997/scan-js-2-clean.zip)
[](url)
tell me how I can let you commit on my branch and it will be done !
@damiencarol

**Comment by valentijnscholten** [comment]
No need for an example, I already specified above what needs to be changed in two files.

>tell me how I can let you commit on my branch and it will be done

Can you check the link I posted above: 

>https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by valentijnscholten** [comment]
Closed in favor of https://github.com/DefectDojo/django-DefectDojo/pull/5004

**Pull request closed without merging**